tests/e2e_regression/run.sh

```diff
@@ -89,7 +89,7 @@ for ((i = 0; i < nCases; i++)); do
       >/tmp/pretty 2>/dev/null &&
       cp /tmp/pretty "$outDir/$name.body" || true
     # Sanitize the current timestamp out of the response header so that diffs are stable
-    sed -i -E 's/^(Date|Content-Length|Last-Modified): .*/\1: UNINTERESTING/g' "$outDir/$name.headers"
+    sed -i '' -E 's/^(Date|Content-Length|Last-Modified): .*/\1: UNINTERESTING/g' "$outDir/$name.headers"
```

Comment:
uhh I remember we've had back-and-forth with `sed -i ''` before :(
This doesn't work on GNU sed (which is what I have locally, and what CI has). It interprets the `''` as the sed script to run. You can try it yourself: `docker run -it bash`, then `echo foo >/tmp/x; sed -i '' 's/o/a/' /tmp/x`.

You could try instead:
- using `echo foo >/tmp/x; sed --in-place='' 's/o/a/' /tmp/x` (though I think that's not supported on MacOS?)
- creating a little helper function that reimplements `sed -i` without using `-i` (*), or using `-i` based on some OS-specific check
- leaving this as-is and using GNU sed on your machine?

(*) something like
```
sed_i() {
    local file="${@: -1}"  # Get the last argument (file path)
    local script="${@: -2:1}" # Get the penultimate argument (sed script)
    local flags="${@:1:$#-2}"  # Get flags (all arguments except the last two)

    local temp_file="$(mktemp)"
    sed "${flags[@]}" "$script" "$file" > "$temp_file"
    mv "$temp_file" "$file"  # Overwrites the original file
}
```

Reply:
@Andrew7234 I think you said you're using GNU sed on your machine currently? So that resolves this thread for now (remember to undo the change), and you can submit the rest of the PR at least. It's useful stuff, would be a shame if it rotted away.